docs/theme/build.ts

```diff
@@ -0,0 +1,245 @@
+        // 'colors' object. This includes core colors defined in tokens/core.json5
+        // and syntax tokens defined in tokens/syntax
+        dictionary.allTokens
+          .filter((token) => {
+            return !['color', 'syntax'].includes(token.path[0]);
+          })
+          .forEach((token) => {
+            // Add each token to the colors object, the name is generated by the custom
+            // transform defined above
+            theme.colors[token.name] = token.value;
+          });
+
+        // Map the syntax styles
+        theme.tokenColors = dictionary.allTokens
+          .filter((token) => {
+            return token.path[0] === 'syntax';
+          })
+          .map((token) => ({
+            scope: token.name,
+            settings: {
+              foreground: token.value,
+              fontStyle: token.fontStyle,
+            },
+          }));
+
+        // Style Dictionary formats expect a string that will be then written to a file
+        return JSON.stringify(theme, null, 2);
+      },
+    },
+    filters: {
+      css: (token) => {
+        return token.path[0] === 'color';
+      },
+    },
+  },
+});
+
+['dark', 'light'].forEach(async (mode) => {
+  await sd
+    .extend({
+      tokens: mode === 'light' ? lightTokens : darkTokens,
+
+      platforms: {
+        vscode: {
+          // Directory to build files to
+          buildPath: `docs/theme/dist/`,
+          // Adding a custom attribute on the platform so we can use it in
+          // the custom format
+          themeType: mode,
+          // The name of the custom transform we defined above
+          transforms: [`color`, `vsCodeName`],
+          files: [
+            {
+              // The path the file will be created at. Make sure this matches
+              // the file paths defined in the package.json
+              destination: `${mode}.vscode.json`,
+              // The name of the custom format defined above
+              format: `vsCodeTheme`,
+            },
+            {
+              // The path the file will be created at. Make sure this matches
+              // the file paths defined in the package.json
+              destination: `${mode}.monaco.json`,
+              // The name of the custom format defined above
+              format: `monaco`,
+            },
+          ],
+        },
+        css: {
+          buildPath: `docs/theme/dist/`,
+          transforms: ['attribute/cti', 'name/kebab', 'cssColor'],
+          prefix: 'sl',
+          files: [
+            {
+              destination: `${mode}.variables.css`,
+              format: 'css/variables',
+              filter: (token) => {
+                // light mode don't include the core colors
+                // because default is dark mode which will have the core colors
+                if (mode === 'light') {
+                  return token.path[0] === 'color' && token.path[1] !== 'core';
```

Comment:
Generally as a best practice I'd personally advise against relying on the `token.path` property.

I think `token.$type` is more reliable and explicit for determining type. For whether it's a core token or not, maybe we should consider putting all core tokens in a folder so we can use `token.filePath` to know whether some token is a core token or not.

I see the `token.path` being relied on in quite a few places here and it reminds me a bit of relying on CTI structure (which is just syntactic sugar on top of the token path), which I kind of wanted to get away from in v4 and in light of DTCG spec. I do see the convenience and making lots of token "attributes" implicit through its token taxonomy/hierarchy, but in my experience it makes things a bit brittle when restructuring tokens.

Reply:
I'm actually using path specifically, and not trying to use the token type (because all the tokens are actually color type). So how would you target tokens based on the dictionary object structure if not by `token.path` ? Or would you re-organize the dictionary object?

Reply:
> how would you target tokens based on the dictionary object structure if not by `token.path` ?

Well, I wouldn't, because imo it's a bad practice to rely upon that, since the structure can be changed and it wouldn't be obvious that that would break your transforms. I would use `token.$type` and `token.filePath` instead, or some metadata prop that lives in the tokenset at the top but does not have a value prop so SD wouldn't classify it as a token, it would just be some metadata living there.

Reply:
Couldn't the same be said about `token.filePath` ? They are both paths, if I change a file name that would break my transforms too. I still don't really see why using the path is bad practice. Also, for this specifically filePath can't be used because the tokens are added via `tokens` and not `source` or `include`. Do you have a specific example you can point to for the best practice?

Reply:
I see your point, they're similar in that the connection between your tokens structure/categorization and your tokens transforms logic isn't explicit and easy to forget about, causing something to break that you didn't expect.

It would be good to think outside the box and see if there's a way we can do it that improves on this problem ("token grouping" that's not tied to token structure/taxonomy), but I agree with you that `token.filePath` isn't that much better than using `token.path`. Feel free to resolve this!